frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -476,7 +478,7 @@ class Visualization extends PureComponent {
             <NoResultsView isSmall={small} />
           ) : error ? (
             <ErrorView
-              error={error}
+              error={errorMessageOverride ?? error}
```

Comment:
Why do we need this? It seems that `error` coming from `props` has precedence anyway: `let error = this.props.error || this.state.error;`

Reply:
Unfortunately that `let error = this.props.error || this.state.error;` is used to determine whether the error state should be displayed or not.

The parent component does not know whether `Visualization` rendering failed and error state should be displayed in it.

In other words: if we used `error` prop instead of `errorMessageOverride`, the error would be always displayed.

Reply:
Got it, makes sense, thanks!